src/Storages/StorageS3.cpp

```diff
@@ -660,8 +660,14 @@ Pipe StorageS3::read(
     Block block_for_format;
     if (isColumnOriented())
     {
+        auto fetch_columns = column_names;
+        fetch_columns.erase(std::remove_if(fetch_columns.begin(), fetch_columns.end(),
+                              [](const String & col){return col == "_path" || col == "_file"; }));
```

Comment:
Maybe it's better to define list of virtual columns somewhere?

Reply:
done